src/Widgets/TerminalWidget.vala

```diff
@@ -174,6 +192,14 @@ namespace PantheonTerminal {
             /* Make Links Clickable */
             this.drag_data_received.connect (drag_received);
             this.clickable (regex_strings);
+
+            GLib.Settings saved_state = new GLib.Settings ("io.elementary.terminal.saved-state");
+            saved_state.bind ("zoom", this, "zoom_factor", GLib.SettingsBindFlags.DEFAULT);
+
+            Idle.add (() => {
+                zoom_factor = zoom_factor;
+                return false;
+            });
```

Comment:
Maybe better to connect to the realize signal to set the font size at the correct time.